### DISCUSSION TITLE
How to get the result of "git describe --tags" by gitlab api?

### DISCUSSION THREAD

**Discussion by skight-shen** [python-gitlab/python-gitlab#2216] [Q&A]
I'd like to get the tag before a specific commit. It's easy by git cmd " git describe --tags".  However, it seems there is no gitlab api which support it.
I  only found the method to get the references the commit has been pushed to by "commit.refs('tag') ": https://python-gitlab.readthedocs.io/en/stable/gl_objects/commits.html.

**Comment by JohnVillalovos** [comment]
I'm not sure. But the place to start looking is probably here:

https://docs.gitlab.com/ee/api/api_resources.html

Now as far as tags specifically there is: https://docs.gitlab.com/ee/api/api_resources.html
Which is python-gitlab is documented here: https://python-gitlab.readthedocs.io/en/main/gl_objects/projects.html#project-tags

My very quick look I didn't find a way to do it via the GitLab API. So you may have to resort to cloning first. I guess it would be possible to iterate over all of the commits and try to figure it out yourself.

**Reply by nejch** [reply]
Yes, I think you'd have to combine listing the commits and finding the refs each of them was pushed to. Not very convenient, I'd do it with git instead of the API.

https://docs.gitlab.com/ee/api/commits.html#list-repository-commits
https://docs.gitlab.com/ee/api/commits.html#get-references-a-commit-is-pushed-to

**Reply by skight-shen** [reply]
I got a work around:
1. get  all the tags order by created_time desc;
2. get the common ancestor by gitlab api:  https://docs.gitlab.com/ee/api/repositories.html#merge-base. once the id in response is equal to the commit combined with the tag, that is the target tag.
